Add workload for odo summit lab

jankleinert opened the pull request:
##### SUMMARY
Adds a workload for the odo summit lab

##### ISSUE TYPE
- New role Pull Request

##### COMPONENT NAME


##### ADDITIONAL INFORMATION

fridim commented on ansible/roles/ocp-workload-odo-workshop/tasks/pre_workload.yml:
```diff
@@ -0,0 +1,11 @@
+    src: ~/.kube/config
+    dest: /tmp/${GUID}
+  environment: 
+    KUBECONFIG: /tmp/${GUID}
```
`${GUID}` is syntax error here. It's jinja template, not Shell.

Plus, environment is useless here.

I would simply add a variable in `defaults/main.yml`:

```yaml
tmp_kubeconfig: /tmp/{{ guid }}
```
Your `copy` task would become:

```yaml
- name: Copy .kube/config and set env var
  copy:
    src: ~/.kube/config
    dest: "{{ tmp_kubeconfig }}"
```
Then you have to add this part:

```
  environment:
    KUBECONFIG: "{{ tmp_kubeconfig }}"
```
to every tasks that run oc commands.

You can factorize using `block` like this:

```yaml
- environment:
    KUBECONFIG: "{{ tmp_kubeconfig }}"
  block:
     - task1
     - task2
     ...
```

fridim commented on ansible/roles/ocp-workload-odo-workshop/tasks/main.yml:
```diff
@@ -0,0 +1,30 @@
+---
+- name: Running Pre Workload Tasks
+  import_tasks: ./pre_workload.yml
```
`./pre_workload.yml` -> can be simply `pre_workload.yml`

fridim commented on ansible/roles/ocp-workload-odo-workshop/tasks/remove_workload.yml:
```diff
@@ -0,0 +1,24 @@
+- name: Remove odo workshop resources
+  shell: "oc delete \
+  services,routes,deploymentconfigs,imagestreams,secrets,configmaps,serviceaccounts,rolebindings,serviceaccounts,rolebindings,persistentvolumeclaims,pods \
+  --selector 'app={{odo_application_name}}-{{odo_project_name}},spawner=learning-portal'"
```
use YAML multiline:

```yaml
- name: Remove odo workshop resources
  shell: >-
    oc delete
    services,routes,deploymentconfigs,imagestreams,secrets,configmaps,serviceaccounts,rolebindings,serviceaccounts,rolebindings,persistentvolumeclaims,pods
    --selector 'app={{odo_application_name}}-{{odo_project_name}},spawner=learning-portal'
```

jankleinert replied:
Thanks! I didn't even know that was possible.

fridim commented on ansible/roles/ocp-workload-odo-workshop/tasks/remove_workload.yml:
```diff
@@ -0,0 +1,24 @@
+- name: Remove odo workshop cluster resources
+  shell: "oc delete \
+  clusterrolebindings,clusterroles \
+  --selector 'app={{odo_application_name}}-{{odo_project_name}},spawner=learning-portal'"
```
same, multiline yaml `>-` is handy

fridim commented on ansible/roles/ocp-workload-odo-workshop/tasks/workload.yml:
```diff
@@ -0,0 +1,21 @@
+   --param PROJECT_NAME={{odo_project_name}} \
+   --param APPLICATION_NAME={{odo_application_name}} \
+   --param TERMINAL_IMAGE={{odo_terminal_image}} \
+   --namespace {{odo_project_name}}"
```
multiline yaml `>-` candidate

fridim commented on ansible/roles/ocp-workload-odo-workshop/tasks/remove_workload.yml:
```diff
@@ -0,0 +1,24 @@
+
+- name: Remove odo workshop project
+  shell: "oc delete project {{odo_project_name}}"
+  ignore_errors: true
```
i would not put `ignore_errors: true` everywhere

You can check for existence of the project, if it exists, then delete.

jankleinert commented:
@fridim Thanks for looking at this - I'm pretty new to ansible, as you can probably tell! I think I've addressed all your comments. I tested this against a non-shared RHPDS  cluster using become_override=true and against one of the shared clusters without setting become_override to true and it seems to work as expected in both cases.

fridim commented:
merged, thanks!
